fix: update list of dependencies in autoware.repos

bgilby59 opened the pull request:
## Description

Added required dependencies to autoware.repos file to successfully complete edge_auto tutorials without errors.

## Tests performed

Followed edge_auto tutorials to launch camera and lidar perception (lidar was simulated with rosbag). Confirmed that tutorials are completed without error and perception runs with reasonable-looking results.

## Effects on system behavior

edge_auto installation process, particularly the following code, now runs without errors:

```
rosdep update
rosdep install -y -r --from-paths src --ignore-src --rosdistro $ROS_DISTRO

./build.sh
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

drwnz commented on autoware.repos:
```diff
@@ -71,3 +67,39 @@ repositories:
+      type: git
+      url: https://github.com/autowarefoundation/autoware_launch.git
+      version: main
+  external/transport_drivers:
```
This repository already exists in this repos file - duplicates causes issues so please remove this one.

drwnz commented on autoware.repos:
You have added the new repositories with `external/` but other repositories in the repos file do not use this, so I would suggest removing for consistency.

Please see the comment about the duplicated repository. Actually, if you merge from the current main branch I think you will not have to add `transport_drivers` as the branch is correct on main currently (`boost`).

drwnz reviewed (changes requested):
Please see the comments in the file.
Required changes:
1. Merge main into your feature branch
2. Remove `external/` folder
3. Double check for duplicates

bgilby59 commented:
@drwnz 
Updated autoware.repos according to your comments and pushed the changes.

Changes:
1. Merged main into my feature branch
2. Fixed folder naming and repo order to match autoware
3. Removed duplicates repository `transport_drivers`
